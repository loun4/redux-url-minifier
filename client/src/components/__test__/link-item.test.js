
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import LinkItem from '../link-item';

let subject;

beforeEach(() => {
  subject = <LinkItem url="/link/Og" />;
});

test('Render link item', () => {
  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Handle shortURL', () => {
  const component = shallow(subject);
  expect(component.find('a').props().href).toEqual('/link/Og');
});

test('Handle copy', () => {
  const component = shallow(subject);

  jest.useFakeTimers();
  component.find('button').simulate('click');
  expect(component.find('button').text()).toEqual('Copied');
  expect(component.state().isCopied).toEqual(true);

  setTimeout(() => {
    component.update();
    expect(component.find('button').text()).toEqual('Copy');
    expect(component.state().isCopied).toEqual(false);
  }, 1000);
  jest.runAllTimers();
});

