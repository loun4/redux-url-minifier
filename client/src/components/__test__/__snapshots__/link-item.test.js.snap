// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render link item 1`] = `
<div
  className="ui buttons"
>
  <a
    className="ui button"
    href="/link/Og"
    target="_blank"
  >
    <i
      className="external alternate icon"
    />
    /link/Og
  </a>
  <div
    className="or"
  />
  <button
    className="ui vertical green button"
    onClick={[Function]}
  >
    Copy
  </button>
</div>
`;
